core/src/main/scala/chisel3/ElaboratedCircuit.scala

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import firrtl.annotations.Annotation
+import firrtl.ir.{CircuitWithAnnos, Serializer}
+import firrtl.options.{CustomFileEmission, Unserializable}
+import chisel3.experimental.{BaseModule, SourceInfo, UnlocatableSourceInfo}
+import chisel3.experimental.hierarchy.Definition
+import chisel3.internal.firrtl.ir.Circuit
+import chisel3.internal.firrtl.Converter
+
+/** The result of running Chisel elaboration
+  *
+  * Provides limited APIs for inspection of the resulting circuit.
+  */
+// This is an important interface so let's keep it separate from the implementation.
+sealed trait ElaboratedCircuit {
+
+  /** The name of the circuit, also the name of the top public module */
+  def name: String
+
+  /** The circuit as a string of FIRRTL IR
+    *
+    * This will include annotations automatically.
+    * For large circuits (> 2 GiB of text) use [[lazilySerialize]].
+    */
+  def serialize: String
+
+  /** The circuit as a string of FIRRTL IR
+    *
+    * For large circuits (> 2 GiB of text) use [[lazilySerialize]]
+    *
+    * @param annotations alternative annotations to include in the FIRRTL IR.
+    */
+  def serialize(annotations: Iterable[Annotation]): String
```

Comment:
Is this the circuit, annotations, or both?

Reply:
Both, will update all of these ScalaDocs. The pattern for "I want just the circuit" is to call this version but pass Nil